src/jit/codegenlinear.cpp

```diff
@@ -335,6 +328,25 @@ void CodeGen::genCodeForBBlist()
         // as we encounter it.
         CLANG_FORMAT_COMMENT_ANCHOR;
 
+#ifdef DEBUG
+        // Set the use-order numbers for each node.
+        {
+            unsigned useNum = 0;
+            for (GenTree* node : LIR::AsRange(block).NonPhiNodes())
+            {
+                assert((node->gtDebugFlags & GTF_DEBUG_NODE_CG_CONSUMED) == 0);
+
+                node->gtUseNum = -1;
+                for (GenTree* operand : node->Operands())
+                {
+                    assert(operand->gtUseNum == -1);
+                    operand->gtUseNum = useNum;
+                    useNum++;
```

Comment:
Although this works, we may want to set these use numbers during LSRA's traversal of a node's operands to help ensure that the correspondence between a node's use number and the relative order in which it is visited by LSRA is correct. Thoughts?